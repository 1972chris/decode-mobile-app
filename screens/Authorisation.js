import React from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { goToPetitionSummary } from '../application/redux/actions/navigation';


const styles = StyleSheet.create({
  authorisationBox: {
    alignSelf: 'stretch',
    backgroundColor: '#FFF',
    margin: 16,
    marginTop: 50,
    paddingHorizontal: 20,
    paddingVertical: 40,
  },
  button: {
    alignSelf: 'center',
    backgroundColor: 'rgb(0,163,158)',
    borderRadius: 2,
    elevation: 2,
    height: 36,
    paddingHorizontal: 16,
    paddingVertical: 8,
    shadowColor: 'rgba(0, 0, 0, 0.54)',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    width: 250,
  },
  buttonText: {
    alignSelf: 'center',
    color: '#FFF',
    fontSize: 14,
    fontWeight: '500',
  },
  container: {
    backgroundColor: 'rgb(246, 246, 246)',
    flex: 1,
    justifyContent: 'space-between',
  },
  logo: {
    alignSelf: 'center',
    fontFamily: 'Lato-Bold',
    fontSize: 25,
    color: 'rgb(203,73,45)',
    marginBottom: 40,
  },
  id: {
    alignSelf: 'center',
    marginBottom: 16,
    color: 'grey',
    fontSize: 10,
  },
  textParagraph: {
    textAlign: 'center',
    marginBottom: 30,
    paddingHorizontal: 10,
  },
});

class Authorisation extends React.Component {
  static route = {
    navigationBar: {
      backgroundColor: 'white',
      tintColor: 'rgb(0,163,158)',
      title: 'Authorise Connection',
    },
  };

  constructor(props) {
    super(props);
    this.goToPetitionSummary = this.goToPetitionSummary.bind(this);
  }

  goToPetitionSummary() {
    this.props.goToPetitionSummary(this.props.petitionLink);
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.authorisationBox}>
          <Text style={styles.logo}>SECURE PETITIONS</Text>
          <Text style={styles.textParagraph}>would like
            to connect with your DECODE wallet
          </Text>
          <TouchableOpacity
            style={styles.button}
            onPress={this.goToPetitionSummary}
          >
            <Text style={styles.buttonText}>AUTHORISE</Text>
          </TouchableOpacity>
        </View>
        <Text style={styles.id}>Wallet ID: {this.props.walletId}</Text>
      </View>
    );
  }
}

Authorisation.propTypes = {
  goToPetitionSummary: PropTypes.func.isRequired,
  petitionLink: PropTypes.string,
  walletId: PropTypes.string,
};

Authorisation.defaultProps = {
  petitionLink: undefined,
  walletId: '',
};

const mapStateToProps = state => ({
  petitionLink: state.petitionLink.petitionLink,
  walletId: state.wallet.id,
});

const mapDispatchToProps = dispatch => ({
  goToPetitionSummary: (petitionLink) => { dispatch(goToPetitionSummary(petitionLink)); },
});

export default connect(mapStateToProps, mapDispatchToProps)(Authorisation);
