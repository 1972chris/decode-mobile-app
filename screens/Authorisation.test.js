import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import Authorisation from './Authorisation';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

const initialState = {
  petitionLink: {
    petitionLink: undefined,
  },
  wallet: {
    id: 'my-id',
  },
};

it('renders Authorisation component', () => {
  const wrapper = shallow(
    <Authorisation />,
    { context: { store: mockStore(initialState) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});
