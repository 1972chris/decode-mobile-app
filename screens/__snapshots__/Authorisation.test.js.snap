// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Authorisation component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(246, 246, 246)",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFF",
        "margin": 16,
        "marginTop": 50,
        "paddingHorizontal": 20,
        "paddingVertical": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "rgb(203,73,45)",
          "fontFamily": "Lato-Bold",
          "fontSize": 25,
          "marginBottom": 40,
        }
      }
    >
      SECURE PETITIONS
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "marginBottom": 30,
          "paddingHorizontal": 10,
          "textAlign": "center",
        }
      }
    >
      would like to connect with your DECODE wallet
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "rgb(0,163,158)",
          "borderRadius": 2,
          "elevation": 2,
          "height": 36,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
          "shadowColor": "rgba(0, 0, 0, 0.54)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 250,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontSize": 14,
            "fontWeight": "500",
          }
        }
      >
        AUTHORISE
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "grey",
        "fontSize": 10,
        "marginBottom": 16,
      }
    }
  >
    Wallet ID: 
    default id
  </Text>
</View>
`;
